File: go/pkg/models/party.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Party struct {
	ID        uuid.UUID `gorm:"type:uuid;default:uuid_generate_v4();primaryKey" json:"id"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
	LeaderID  uuid.UUID `json:"leaderId"`
	Leader    User      `gorm:"foreignKey:LeaderID" json:"leader"`
	Members   []User    `gorm:"foreignKey:PartyID" json:"members"`
}

func (p *Party) BeforeCreate(tx *gorm.DB) (err error) {
	if p.ID == uuid.Nil {
		p.ID = uuid.New()
	}
	return nil
}

File: dart/sonar_web/lib/models/party.dart
import 'user.dart';

class Party {
  final String id;
  final String createdAt;
  final String updatedAt;
  final String leaderId;
  final User leader;
  final List<User> members;

  const Party({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    required this.leaderId,
    required this.leader,
    required this.members,
  });

  factory Party.fromJson(Map<String, dynamic> json) {
    return Party(
      id: json['id'] as String,
      createdAt: json['createdAt'] as String? ?? '',
      updatedAt: json['updatedAt'] as String? ?? '',
      leaderId: json['leaderId'] as String,
      leader: User.fromJson(json['leader'] as Map<String, dynamic>),
      members: (json['members'] as List<dynamic>?)
              ?.map((e) => User.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
    );
  }
}
